control-plane/pkg/reconciler/kafka/kafka.go

```diff
@@ -0,0 +1,198 @@
+
+import (
+	"fmt"
+	"strconv"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kedav1alpha1 "knative.dev/eventing-autoscaler-keda/third_party/pkg/apis/keda/v1alpha1"
+	"knative.dev/pkg/kmeta"
+	"knative.dev/pkg/reconciler"
+
+	"knative.dev/eventing-autoscaler-keda/pkg/reconciler/keda"
+	kafkainternals "knative.dev/eventing-kafka-broker/control-plane/pkg/apis/internals/kafka/eventing/v1alpha1"
+)
+
+const (
+	defaultKafkaLagThreshold = 10
+)
+
+func GenerateScaleTarget(cg *kafkainternals.ConsumerGroup) *kedav1alpha1.ScaleTarget {
+	return &kedav1alpha1.ScaleTarget{
+		Name:       cg.Name,
+		APIVersion: kafkainternals.SchemeGroupVersion.String(),
+		Kind:       "ConsumerGroup",
+	}
+}
+
+func GenerateScaleTriggers(cg *kafkainternals.ConsumerGroup, triggerAuthentication *kedav1alpha1.TriggerAuthentication) ([]kedav1alpha1.ScaleTriggers, error) {
+	triggers := []kedav1alpha1.ScaleTriggers{}
+	bootstrapServers := cg.Spec.Template.Spec.Configs.Configs["bootstrap.servers"]
+	consumerGroup := cg.Spec.Template.Spec.Configs.Configs["group.id"]
+
+	lagThreshold, err := keda.GetInt32ValueFromMap(cg.Annotations, keda.KedaAutoscalingKafkaLagThreshold, defaultKafkaLagThreshold)
+	if err != nil {
+		return nil, err
+	}
+
+	allowIdleConsumers := "false"
+	if cg.Status.Placements != nil {
+		allowIdleConsumers = "true"
+	}
+
+	for _, topic := range cg.Spec.Template.Spec.Topics {
+		triggerMetadata := map[string]string{
+			"bootstrapServers":   bootstrapServers,
+			"consumerGroup":      consumerGroup,
+			"topic":              topic,
+			"lagThreshold":       strconv.Itoa(int(*lagThreshold)),
+			"allowIdleConsumers": allowIdleConsumers,
+		}
+
+		trigger := kedav1alpha1.ScaleTriggers{
+			Type:              "kafka",
+			Metadata:          triggerMetadata,
+			AuthenticationRef: &kedav1alpha1.ScaledObjectAuthRef{},
+		}
+
+		if triggerAuthentication != nil {
+			trigger.AuthenticationRef.Name = triggerAuthentication.Name
+		}
+
+		triggers = append(triggers, trigger)
+	}
+
+	return triggers, nil
+}
+
+func GenerateTriggerAuthentication(cg *kafkainternals.ConsumerGroup, saslType *string) (*kedav1alpha1.TriggerAuthentication, *corev1.Secret, error) {
+
+	secretTargetRefs := []kedav1alpha1.AuthSecretTargetRef{}
+
+	secret := corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-secret", cg.Name),
+			Namespace: cg.Namespace,
+			OwnerReferences: []metav1.OwnerReference{
+				*kmeta.NewControllerRef(cg),
+			},
+		},
+		Data:       make(map[string][]byte),
+		StringData: make(map[string]string),
+	}
+
+	if cg.Spec.Template.Spec.Auth.NetSpec.SASL.Enable {
+
+		if saslType != nil {
+			switch *saslType {
+			case "SCRAM-SHA-256":
+				secret.StringData["sasl"] = "scram_sha256"
+			case "SCRAM-SHA-512":
+				secret.StringData["sasl"] = "scram_sha512"
+			case "PLAIN":
+				secret.StringData["sasl"] = "plaintext"
+			default:
+				return nil, nil, fmt.Errorf("SASL type value %q is not supported", *saslType)
```

Comment:
isn't this done similar already somewhere else?

Reply:
Yes, in EAK for KafkaSource [here](https://github.com/knative-sandbox/eventing-autoscaler-keda/blob/main/pkg/reconciler/kafka/kafka.go#L102-L117). Hopefully we can get rid of that "eventually" once we have KEDA scaling for Kafka fully supported in this repo...